github/repos.go

```diff
@@ -334,6 +338,36 @@ func (s *RepositoriesService) Create(ctx context.Context, org string, repo *Repo
 	return r, resp, nil
 }
 
+// TemplateRepoRequest represents a request to create a repo from a template
+type TemplateRepoRequest struct {
+	// Name is required when creating a repo.
+	Name        *string `json:"name,omitempty"`
+	Owner       *string `json:"owner,omitempty"`
+	Description *string `json:"description,omitempty"`
+
+	Private *bool `json:"private,omitempty"`
+}
+
+// Generate a repository using a repository template
```

Comment:
Hmmm... I see that GitHub named this endpoint `generate`, which seems odd to me.

@gauntface - what do you think about calling this endpoint `repositories.CreateFromTemplate` (or maybe just `FromTemplate`?) as that seems to more appropriate to me?

Either way, @shanempope, we will want to make this comment a complete sentence for the auto-generated godocs.  Hopefully something like:

`// CreateFromTemplate generates a repository from a template.`